### PR TITLE
fix(report): coverage update for xsl:assert

### PR DESCRIPTION
**Pull request by galtm**
This pull request marks xsl:assert and all its descendants as "unknown" except where certain higher-priority templates apply. This pull request also adds some test cases that illustrate the xsl:assert rule being lower priority than another template.

Note: This pull request does not include the changes in the open pull request https://github.com/xspec/xspec/pull/1978.

Fixes https://github.com/xspec/xspec/issues/1940.

---

Cc: @birdya22

### CODE REVIEW

**Comment by galtm** [comment]
Oops, I mentioned the wrong element in the pull request title at first. Fixed now! This PR is about xsl:assert.

File: src/reporter/coverage-compute-status.xsl
Comment (birdya22) [inline src/reporter/coverage-compute-status.xsl:1]: A general comment, not specifically related to this change.
For the benefit of people looking at this file in the future, could you add some comments, probably near the `mode="coverage"` section, that these templates relate to the rules in `xslt-code-coverage-by-element.md` and also something generally about the use of `priority` values to ensure the correct template is matched..

Comment (galtm) [reply src/reporter/coverage-compute-status.xsl:1]: Sure. How about this, where the 2nd and 3rd paragraph are new?
```
    <!--
      mode="coverage"
      
      This mode implements the rules described in docs/xslt-code-coverage-by-element.md.
      
      Priority values of templates in this mode ensure correct rule selection, such as
      when templates match descendants via different paths. Tests for rule selection are
      in test/end-to-end/cases-coverage.
   -->
    <xsl:mode name="coverage" on-multiple-match="fail" on-no-match="fail" />
```

Comment (galtm) [reply src/reporter/coverage-compute-status.xsl:1]: 74ac4ca91b148b6d03d7d0924d3736cdd054776e adds those paragraphs to the comment.

**Pull request merged**